### PR TITLE
chore: release forest v0.6.0

### PR DESCRIPTION
**Pull request by jdjaustin**
**Summary of changes**
Changes introduced in this pull request:
- Summarize the changes in the upcoming 0.6.0 release.

**Reference issue to close (if applicable)**

N/A

**Other information and links**

N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,46 @@
...
+* Multiple CI improvements.
+
+### Removed
+* All calls to `forest_statediff` in daemon.
+* `std sleep` and cargo check for slow tests.
+* Unused `ProofVerifier` trait.
+* RocksDB check for low file descriptor limit.
+* Miner-specific dead code.
+* Unused RPC endpoints.
+* `funty` crate patch.
```
Comment (LesnyRumcajs) [inline CHANGELOG.md:43]: We should list the changes that actually impact the Forest usage. Internals are not helpful to end users and they kind of bloat the changelog.
```suggestion
* RocksDB check for low file descriptor limit.
* Unused RPC endpoints.
```

**Review by LesnyRumcajs — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,41 @@
-## Forest v0.6.0 (unreleased, scheduled for 2022-12-30)
+## Forest v0.6.0 (2023-01-04)
+
+Notable updates:
+* Added support for the new Protocol Labs snapshot service.
+* Implemented a database benchmarking script to quantify baseline performance against ParityDb.
+* Several improvements to logging (including integration with Grafana Loki) and error handling.
+
+### Added
+* New daemon option flag `—log-dir` for log file support.
```
Comment (lemmih) [inline CHANGELOG.md:9]: I think your editor changed `--` to `—`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,41 @@
...
+* Allow `—height` flag to take relative offsets for validating the tipsets in a snapshot.
+* Propagate fatal errors from background tasks.
+* Improved error messages when signing and verifying messages.
+* Fixed issue with invalid snapshot exports.
+* Updated `snapshot fetch` subcommands to support the new Protocol Labs snapshot service.
+* Improved caching of cargo-installed bins.
+* Replaced unmaintained `difference` with `similar`.
+* Fixed RPC `net disconnect` endpoint.
+* Corrected RPC serialization of FIL balances.
+* Multiple CI improvements.
```
Comment (lemmih) [inline CHANGELOG.md:34]: The target audience of the changelog are users, not developers. They only care about user-visible changes to Forest.

- They don't care about the DB benchmark script as it is an internal development tool. They don't care which module we use to serialize policies. They _might_ care about the logging framework but that is also borderline.
- The sync issue description is too detailed. It should be much simpler. Something like: `Fixed bug triggered by large network messages.`
- The same goes for the missing cursor. Highlight that it is a bug-fix and keep it short: "Fixed bug where cursor would be hidden after keyboard interrupt."
- Users don't care about `serde_ipld_dagcbor`.
- There are three or four items mentioning improved error messages. They can be merged into one item.
- The caching of cargo bins isn't a user visible change.
- Replacing `difference` with `similar` isn't a user visible change.
- The `net disconnect` bug could be explained in a bit more detail.
- The FIL balance bug fix could be explained in a bit more detail.
- The CI improvements are not user visible.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,32 @@
-## Forest v0.6.0 (unreleased, scheduled for 2022-12-30)
+## Forest v0.6.0 (2023-01-04)
+
+Notable updates:
+* Added support for the new Protocol Labs snapshot service.
+* Several improvements to logging (including integration with Grafana Loki) and error handling.
+
+### Added
+* New daemon option flag `-—log-dir` for log file support.
```
Comment (lemmih) [inline CHANGELOG.md:8]:
```suggestion
* New daemon option flag `--log-dir` for log file support.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,32 @@
...
+* New ParityDb section in configuration (including statistics and compression settings).
+* Integration with Grafana Loki for more advanced log filtering and summarization.
+* Peer tipset epoch now in metrics.
+
+### Changed
+* Several improvements to error handling.
+* Implemented proper tagging strategy for Docker images.
+* Restored link to [deps.rs](https://deps.rs/).
```
Comment (lemmih) [inline CHANGELOG.md:16]: The link was removed after `0.5.1` so this line shouldn't be here.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,32 @@
...
+* Upgraded toolchain channel to `nightly-2022-12-15`.
+* Migrated logging infrastructure from `log` to `tracing`.
+* Fixed bug triggered by large network messages.
+* Fixed bug where cursor would be hidden after keyboard interrupt.
+* Allow `-—height` flag to take relative offsets for validating the tipsets in a snapshot.
+* Fixed issue with invalid snapshot exports.
```
Comment (lemmih) [inline CHANGELOG.md:22]: What issue? When would this happen?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,32 @@
...
+
+### Added
+* New daemon option flag `-—log-dir` for log file support.
+* New ParityDb section in configuration (including statistics and compression settings).
+* Integration with Grafana Loki for more advanced log filtering and summarization.
+* Peer tipset epoch now in metrics.
+
+### Changed
+* Several improvements to error handling.
+* Implemented proper tagging strategy for Docker images.
```
Comment (lemmih) [inline CHANGELOG.md:15]: We can be more explicitly about what "proper tagging" means:

```suggestion
* Docker images are now tagged with version (eg `0.6.0`) and date (eg `2023-01-05-commit_hash`).
```

**Review by lemmih — APPROVED** [review]
I like it. Remember to update the release date (current set to `2023-01-04`) before merging.

Also, could you wrap the lines at 80 or 100 characters. Your editor will have a command for doing this.

**Pull request merged**